### PR TITLE
Closes #11 by implementing isAnything() as a core matcher. The dummy …

### LINKED ISSUE #11: New Matcher: isAnything()
*Request:* Add an `isAnything()` matcher that will match anything. This won't have much use in direct matches. But when a matcher is used to define an expected parameter in a mocking framework, it becomes useful when you are defining params where you don't care what the value is. For example:
```
mock.given("fooFunction").with(["foo", isAnything(), true]).willReturn("bar")
```

### PR DESCRIPTION
**Pull request by imbenjamin**
…matcher returns true and is also capable of handling an optional parameter

### CODE REVIEW

File: source/brsHamcrest_Matchers/brsHamcrest_CoreMatchers.brs [hunk 1]
```diff
@@ -15,6 +15,26 @@ function is (matcher as Object) as Object
...
 
 
+'A matcher that has the correct structure, but returns true to anything, for the purposes of integration with frameworks
+'Is capable of handling an optional parameter
+'
+'Example:
+'assertThat(foo, isAnything())
+'
+'@return {Object<Matcher>} A Matcher
+function isAnything (optionalValue="" as Dynamic) as Object
```
Comment (oneronaut) [inline source/brsHamcrest_Matchers/brsHamcrest_CoreMatchers.brs]: Why do you need an optional value here?

**Review by oneronaut — CHANGES_REQUESTED** [review]

File: source/brsHamcrest_Matchers/brsHamcrest_CoreMatchers.brs (hunk 1 above)
Comment (imbenjamin) [reply source/brsHamcrest_Matchers/brsHamcrest_CoreMatchers.brs]: Not sure what my thinking was at the time, perhaps I was just wanted to be as open as possible, I'll remove it.

**Comment by imbenjamin** [comment]
@oneronaut Resolved conflicts, please review again.

File: tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_CoreMatchers.brs [hunk 2]
```diff
@@ -212,6 +212,23 @@ sub test_isNot_arrayNoneTrue (t as Object)
...
+    'WHEN'
+    result = isAnything().doMatch(knownTarget)
+
+    'THEN'
+    t.assertTrue(result)
+
+    teardown_brsHamcrest_CoreMatchers()
+end sub
+
+
```
Comment (oneronaut) [inline tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_CoreMatchers.brs:231]: I would add a few more unit tests that confirm `isAnything()` really does match anything.

Comment (oneronaut) [reply tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_CoreMatchers.brs:231]: Perhaps another time...

**Review by oneronaut — APPROVED** [review]

**Pull request merged**